### DISCUSSION TITLE
Controller performs the resize but doesn't log it properly.

### DISCUSSION THREAD

**Discussion by umeshbitgo** [topolvm/pvc-autoresizer#291] [Q&A] [answered]
pvc-autoresizer resizes the PVCs when they hit the threshold, when I follow the log it says `resize started` from current size to the next size as per the annotation. It never says that a PVC has been resized, it keeps saying `waiting for resize` but completes the resizing and doesn't log the event. I looked at the source code and it should log when a PVC is successfully resized. am I missing something?
<img width="1309" alt="Screenshot 2024-11-20 at 10 08 27 AM" src="https://github.com/user-attachments/assets/e1f1999a-7f8a-49b7-8a3b-e69bff06749f">

**Comment by ushitora-anqou** [comment, marked as answer]
pvc-autoresizer doesn't log a success of resize because it doesn't directly detect it. By [this `if` statement](https://github.com/topolvm/pvc-autoresizer/blob/f794d6fe38804d77954a6b02fa21d472a5b34401/internal/runners/pvc_autoresizer.go#L203), it just checks whether the current size of the volume is stable (i.e., not in the process of resizing) or still undergoing resizing. Actually, it can't detect the success because there's no specific indication that an reconciliation iteration is the first one following a successful resize.

**Comment by ushitora-anqou** [comment]
This discussion has been closed due to inactivity. Please feel free to reopen this discussion (or open a new one) if you still need to discuss this topic. Thank you for your contribution.
